### PR TITLE
updated webpack-dev-server to accept a custom ssl certificate

### PR DESCRIPTION
**Pull request by polco**
Please fill in this template.

- [x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [x ] Test the change in your own code. (Compile and run.)
- [ x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by plantain-00 — APPROVED** [review]

File: types/webpack-dev-server/index.d.ts [hunk 1]
```diff
@@ -13,7 +13,11 @@ declare namespace WebpackDevServer {
     interface Configuration {
         contentBase?: string;
         hot?: boolean;
-        https?: boolean;
+        https?: boolean | {
+            key: string,
```
Comment (daveparslow) [inline types/webpack-dev-server/index.d.ts]: Doesn't the following line imply that key and cert are optional (i.e. will use a fake cert)
https://github.com/webpack/webpack-dev-server/blob/97484a9cebad152e67b2fa5ca40058962295675c/lib/Server.js#L463

Comment (daveparslow) [reply types/webpack-dev-server/index.d.ts]: Also can you add the rest of the params?
https://github.com/webpack/webpack-dev-server/blob/97484a9cebad152e67b2fa5ca40058962295675c/lib/Server.js#L375

**Review by daveparslow — CHANGES_REQUESTED** [review]
Can you check if the parameters are optional?  Would be nice to add the rest of the parameters for the https section.

**Comment by polco** [comment]
@daveparslow turns out webpack-dev-server is just feeding the https object to spdy. So i just use the ServerOptions type of spdy instead.

server creation call:
https://github.com/webpack/webpack-dev-server/blob/97484a9cebad152e67b2fa5ca40058962295675c/lib/Server.js#L472
spdy type: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/spdy/index.d.ts#L85

**Review by daveparslow — APPROVED** [review]

**Pull request merged**